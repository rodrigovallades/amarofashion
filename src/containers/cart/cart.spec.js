import React from 'react';
import { shallow } from 'enzyme';
import immutable from 'immutability-helper';

import { Cart } from './index';

const mockToggle = jest.fn();

const props = {
	toggle: mockToggle,
	cart: {
		isActive: false,
		data: [],
	},
	fullscreen: false,
}

function setup(ownProps = props) {
  return shallow(<Cart {...ownProps} />);
}

describe('Cart', function () {
	let wrapper;

	describe('with cart closed', () => {
		beforeAll(() => {
			wrapper = setup();
		});

		it('should render toggle icon', () => {
			expect(wrapper.find('.cart__toggler').exists()).toBe(true);
		});
	});

	describe('with cart open', () => {
		beforeAll(() => {
			wrapper.setProps(immutable(wrapper.instance().props, {
				cart: {
					isActive: { $set: true },
				},
			}));
		});

		it('should render products and sum', () => {
			expect(wrapper.find('.cart__container').exists()).toBe(true);
			expect(wrapper.find('.cart__total').exists()).toBe(true);
			expect(wrapper.find('.cart__container').exists()).toBe(true);
		});
	});
})
