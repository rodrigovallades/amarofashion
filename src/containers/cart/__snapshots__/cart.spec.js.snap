// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart container renders "Cart" deep correctly 1`] = `
<div>
  <div
    className="cart__toggler"
    onClick={[Function]}
  >
    <div>
      <span
        className="cart__toggler-badge"
      >
        0
      </span>
    </div>
  </div>
</div>
`;
